vagrant/provisioning/roles/upgrade-solr/tasks/main.yml

```diff
@@ -1,12 +1,17 @@
-- name: Check systemd zookeeper file exists
-  stat:
-    path: /etc/systemd/system/zookeeper.service
-  register: service_file
+- name: Populate service facts
+  service_facts:
 
 - name: install zookeeper
   include_role: 
     name: zookeeper
-  when: not service_file.stat.exists
+  when: ansible_facts.services["zookeeper.service"] is not defined
```

Comment:
@david-oc-miller You are right, this is better approach. Plus, i added extra steps, just for safety if zookeeper is there, but not running. Since we need him to write configuration for Solr with zkcli.